File: Temperatures.java
// Time Complexity :O(n) where n is length of the temperatures array
// Space Complexity :O(n) where n is length of the temperatures array
// Did this code successfully run on Leetcode :yes
// Any problem you faced while coding this :
import java.util.Stack;
public class Temperatures {
    public int[] dailyTemperatures(int[] temperatures) {
        if(temperatures == null || temperatures.length < 1){
            return null;
        }
        int len = temperatures.length;
        int[] result = new int[len];

        Stack<Integer> st = new Stack<Integer>();
        for(int i=0; i<len; i++){
            while(!st.isEmpty() && temperatures[i] > temperatures[st.peek()]){
                int popped = st.pop();
                result[popped] = i-popped;
            }
            st.push(i);
        }
        return result;
    }
}

File: temperatures.py
#TC = O(n)
#SC = O(n)
# Did it work in leetcode: Yes

class Solution:
    
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        if len(temperatures) == 0:
            return result
        stack = []
        n=len(temperatures)
        result = [0] * n
        
        for i in range(n):
            while len(stack) != 0 and temperatures[i] > temperatures[stack[-1]]:
                popped = stack.pop()
                result[popped] = i - popped
            
            stack.append(i)
        return result
            
            
        